lib/controller/mixins/next-step.js

```diff
@@ -82,6 +82,10 @@ module.exports = Controller => class extends Controller {
             });
         }
 
+        if (typeof nextStep === 'function') {
+            nextStep = nextStep(req, res);
```

Comment:
can you run it in the context of the controller, as that allows calling controller methods.